File: CaseStudies/glass/Implementations/C/src/Control.c
#include "InputParameters.h"
#include "InputFormat.h"
#include "DerivedValues.h"
#include "InputConstraints.h"
#include "ReadTable.h"
#include "Calculations.h"
#include "ArrayLib.h"
#include "Constants.h"
#include "OutputFormat.h"

#include <stdio.h>

int main(int argc, char* argv[]){
  InputParameters* params = new_InputParameters();
  get_input(argv[1], params);
  derived_params(params);
  check_constraints(params);
  
  double* w_array;
  int w_array_n = read_table_row(double)("TSD.txt", 0, 1, 0, 2, ',', &w_array);
  
  double** data_sd;
  double** data_q;
  int data_sd_n = read_table(double)("TSD.txt", 1, 0, w_array_n*2, 2, ',', &data_sd);
  int data_q_n = read_table(double)("TSD.txt", 1, 1, w_array_n*2, 2, ',', &data_q);
  
  double* j_array;
  int j_array_n = read_table_row(double)("SDF.txt", 0, 1, 0, 2, ',', &j_array);
  
  double** data_asprat;
  double** data_qstar;
  int data_asprat_n = read_table(double)("SDF.txt", 1, 0, w_array_n*2, 2, ',', &data_asprat);
  int data_qstar_n = read_table(double)("SDF.txt", 1, 1, w_array_n*2, 2, ',', &data_qstar);
  
  double q = calc_q(w_array, w_array_n, data_sd, data_sd_n, data_q, data_q_n, params);
  
  double* q_vec;
  int q_vec_n = calc_q_vec(j_array, j_array_n, data_asprat, data_asprat_n, data_qstar, params, &q_vec);
  
  double* j_vec;
  int j_vec_n = calc_j_vec(j_array, j_array_n, data_asprat, data_asprat_n, data_qstar, params, &j_vec);

  double q_hat = calc_q_hat(q, q_vec, q_vec_n, params);
  double j_tol = calc_j_tol(params);
  double j = calc_j(q_vec, q_vec_n, j_vec, q_hat);
  double q_hat_tol = calc_q_hat_tol(q_vec, j_vec, j_vec_n, j_tol);
  double pb = calc_pb(j, params);
  double nfl = calc_nfl(q_hat_tol, params); 
  double lr = calc_lr(nfl, params);
  int is_safe1 = calc_is_safe1(pb, params);
  int is_safe2 = calc_is_safe2(lr, q);
  
  display_output("outputfile.txt", q, j, q_hat_tol, pb, lr, nfl, is_safe1, is_safe2, params);
  
}

File: Presentations/WG2_11/GlassBR_Code/fromGOOL/csharp/Control.cs
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

namespace GlassBR {
    public class Control {
        
        public static void Main(string[] args) {
            string filename = args[0];
            InputParameters inparams = new InputParameters();
            InputFormat.get_input(filename, inparams);
            DerivedValues.derived_params(inparams);
            InputConstraints.check_constraints(inparams);
            List<double> w_array = ReadTable.read_z_array("TSD.txt");
            List< List<double> > data_sd = ReadTable.read_x_array("TSD.txt");
            List< List<double> > data_q = ReadTable.read_y_array("TSD.txt");
            List<double> j_array = ReadTable.read_z_array("SDF.txt");
            List< List<double> > data_asprat = ReadTable.read_x_array("SDF.txt");
            List< List<double> > data_qstar = ReadTable.read_y_array("SDF.txt");
            double q = Interpolation.interpY(data_sd, data_q, w_array, inparams.sd, inparams.wtnt);
            double q_hat = Calculations.calc_q_hat(q, inparams);
            double j_tol = Calculations.calc_j_tol(inparams);
            double j = Interpolation.interpZ(data_asprat, data_qstar, j_array, inparams.asprat, q_hat);
            double q_hat_tol = Interpolation.interpY(data_asprat, data_qstar, j_array, inparams.asprat, j_tol);
            double pb = Calculations.calc_pb(j, inparams);
            double nfl = Calculations.calc_nfl(q_hat_tol, inparams);
            double lr = Calculations.calc_lr(nfl, inparams);
            Boolean is_safe1 = Calculations.calc_is_safe1(pb, inparams);
            Boolean is_safe2 = Calculations.calc_is_safe2(lr, q);
            OutputFormat.display_output("outputfile.txt", q, j, q_hat_tol, pb, lr, nfl, is_safe1, is_safe2, inparams);
            Console.WriteLine("Main has been executed and the results have been written to 'outputfile.txt'.");
        }
    }
}

